### DISCUSSION TITLE
当目录名和文件名相同的时候会死循环

### DISCUSSION THREAD

**Discussion by youyifentian** [messense/aliyundrive-webdav#658] [Q&A]
### 问题描述

使用场景群晖CloudSync使用aliyunwebdav同步本地到阿里云
1、比如本地目录：/volume1/cloudreve/cloudreve  该目录的第一个cloudreve是文件夹，第二个cloudreve是文件，
2、比如本地目录：/volume2/aliyunwebdav/1.8.9/aliyunwebdav  该目录的第一个aliyunwebdav是文件夹，第二个aliyunwebdav是文件
这样同步到阿里云的时候CloudSync就会死循环，反复同步，查看阿里云网页就发现回收站出现多个cloudreve和aliyunwebdav的删除文件夹，会反复创建和删除目录路径里面文件夹和文件同名的文件夹，包括更深层的文件和上层父文件夹同名的时候也会出现
修改文件夹名字后可恢复正常，比如例1中的/volume1/cloudreve/cloudreve修改为/volume1/123cloudreve/cloudreve即可正常同步

### 重现步骤

1、正常设置aliyundrive-webdav并启动
2、创建父文件夹和子文件同名的测试目录，
比如创建目录：/volume1/test/，注意test目录为稍后同步目录，
在test目录下创建cloudreve为文件夹，该文件夹下有一名为cloudreve的文件，另在test目录下创建aliyunwebdav文件夹，并在该文件夹下创建名为aliyunwebdav的文件，最终目录结构如下：
目录：/volume1/test/
目录：/volume1/test/cloudreve
目录：/volume1/test/aliyunwebdav
文件：/vloume1/test/cloudreve/cloudreve
文件：/volume1/test/aliyunwebdav/aliyunwebdav
3、打开群晖CloudSync套件，创建webdav同步，本地目录选择：/volume1/test，远程目录选择根目录下的任意文件夹即可，同步方向选择仅上传本地更改，保存即可
4、等待上传，即会出现cloudreve和aliyunwebdav反复同步死循环，阿里云反复创建、删除这两个文件夹

### 版本

1.10.1

### 运行平台

群晖 x86_64

### refresh token 获取方式

网页、APP均试过

### 日志

忘了记录日志了

**Comment by jazzychai** [comment]
我用群晖上传东西传了三天没传完我才发现。。。也是这个问题

**Comment by tmzg0000** [comment]
我也是用群晖发生同样的问题。

**Comment by blackshow** [comment]
+1

**Comment by messense** [comment]
没有群晖，帮不了你。
